optuna/storages/_rdb/storage.py

```diff
@@ -81,6 +84,25 @@ def _create_scoped_session(
             )
         else:
             raise
+    except sqlalchemy_exc.OperationalError as e:
+        session.rollback()
+
+        # The following keystrings depend on the backend DB.
+        # If another issue about another DB is raised, the keystrings will be updated.
+        error_keystrings = ("1205", "Lock wait timeout exceeded; try restarting transaction")
```

Comment:
This is very fragile to version differences as well, so I think we need to discuss a bit more.

Reply:
Agreed. I will remove this condition. Arbitrary `OperationalError` will be catched and retried. In other words, I will assume that errors that retry has no effect will not occur.